src/main/java/com/app/onlinebookstore/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,59 @@
+package com.app.onlinebookstore.controller;
+
+import com.app.onlinebookstore.dto.CartItemDto;
+import com.app.onlinebookstore.dto.CreateCartItemRequestDto;
+import com.app.onlinebookstore.dto.ShoppingCartDto;
+import com.app.onlinebookstore.dto.UpdateCartItemDto;
+import com.app.onlinebookstore.service.CartItemService;
+import com.app.onlinebookstore.service.ShoppingCartService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping cart management", description = "Endpoints for managing shopping cart")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/api/cart")

```

Comment:
You can add "/api" path to application properties and don't write it in every controller mapping : `server.servlet.context-path=/api
`